lib/charms/data_platform_libs/v0/object_storage.py

```diff
@@ -0,0 +1,262 @@
+        super().__init__(
+            model,
+            relation_name,
+        )
+        self.container = container
+
+
+class AzureStorageRequirerEventHandlers(RequirerEventHandlers):
+    """Event handlers for for requirer side of Azure Storage relation."""
+
+    on = AzureStorageRequirerEvents()  # pyright: ignore[reportAssignmentType]
+
+    def __init__(
+        self, charm: CharmBase, relation_data: AzureStorageRequirerData
+    ):
+        super().__init__(charm, relation_data)
+
+        self.relation_name = relation_data.relation_name
+        self.charm = charm
+        self.local_app = self.charm.model.app
+        self.local_unit = self.charm.unit
+
+        self.framework.observe(
+            self.charm.on[self.relation_name].relation_joined, self._on_relation_joined_event
+        )
+        self.framework.observe(
+            self.charm.on[self.relation_name].relation_changed, self._on_relation_changed_event
+        )
+
+        self.framework.observe(
+            self.charm.on[self.relation_name].relation_broken,
+            self._on_relation_broken_event,
+        )
+
+    def _on_relation_joined_event(self, event: RelationJoinedEvent) -> None:
+        """Event emitted when the Azure Storage relation is joined."""
+        logger.info(f"Azure storage relation ({event.relation.name}) joined...")
+        if self.container is None:
+            self.container = f"relation-{event.relation.id}"
+        event_data = {"container": self.container}
+        self.relation_data.update_relation_data(event.relation.id, event_data)
+
+    def get_azure_connection_info(self) -> Dict[str, str]:
+        """Return the azure storage connection info as a dictionary."""
+        for relation in self.relations:
+            if relation and relation.app:
+                return self.relation_data.fetch_relation_data([relation.id])[relation.id]
+        return {}
+
+    def _on_relation_changed_event(self, event: RelationChangedEvent) -> None:
+        """Notify the charm about the presence of Azure credentials."""
+        logger.info(f"Azure storage relation ({event.relation.name}) changed...")
+
+        diff = self._diff(event)
+        if any(newval for newval in diff.added if self.relation_data._is_secret_field(newval)):
+            self.relation_data._register_secrets_to_relation(event.relation, diff.added)
+
+        # check if the mandatory options are in the relation data
+        contains_required_options = True
+        credentials = self.get_azure_connection_info()
+        missing_options = []
+        for configuration_option in AZURE_STORAGE_REQUIRED_INFO:
+            if configuration_option not in credentials:
+                contains_required_options = False
+                missing_options.append(configuration_option)
+
+        # emit credential change event only if all mandatory fields are present
+        if contains_required_options:
+            getattr(self.on, "credentials_changed").emit(
```

Comment:
If I'm correct, at this point the credentials did NOT change.

IT's the connection information that changed.

If the credentials change, it's a `seceret-changed` event that was triggered. Not a `relation-changed` event.

I'm suggesting to introduce a `storage_connection_info_changed` event instead of using an event suggesting credentials change.

Reply:
I'll rename the events as `storage_connection_info_changed` and `storage_connection_info_requested`.